commit fc4381d7c3e8d7440a3e33430d24ebf217a99142
Author: Vladislav Zavjalov <slazav@altlinux.org>
Date:   Thu Oct 2 12:24:04 2008 +0400

    fix calling null function (todo: problem with initialize_char_handler after resize?)

Fixes crash. To reproduce create some object, zoom in so that it does not fit
the screen, start another edit and press Ctrl+Z.

diff --git a/xfig/u_redraw.c b/xfig/u_redraw.c
index 146abfa..fd53a8e 100644
--- a/xfig/u_redraw.c
+++ b/xfig/u_redraw.c
@@ -434,7 +434,7 @@ redisplay_curobj()
       }
     } else {
 	/* editing an object, just refresh it as is */
-	(*canvas_ref_proc)(cur_x, cur_y);
+	if (canvas_ref_proc) (*canvas_ref_proc)(cur_x, cur_y);
     }
 }
 
